File: payment/src/main/java/com/raju/karthikeyan/payment/di/PaymentUiModule.java
package com.raju.karthikeyan.payment.di;

import com.raju.karthikeyan.payment.ICheckoutManager;
import com.raju.karthikeyan.payment.PaymentManager;
import com.raju.karthikeyan.payment.PaymentRepository;
import com.raju.karthikeyan.payment.PaymentUseCase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PaymentUiModule {
    @Provides
    @Singleton
    PaymentManager providePaymentManager(ICheckoutManager checkoutManager){
        return new PaymentManager(checkoutManager);
    }
}

File: payment/src/main/java/com/raju/karthikeyan/payment/di/PaymentUiModule.kt
package com.raju.karthikeyan.payment.di

import com.raju.karthikeyan.payment.PaymentManager
import org.koin.core.qualifier.named
import org.koin.dsl.module

val paymentUiModule = module {

    scope(named(name = "PaymentUIScope")) {

        scoped { PaymentManager(get()) }
    }
}